src/3p-analytics-common.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @const {string} */
+export const AMP_ANALYTICS_3P_EVENT_MESSAGES_TYPE = 'AA3pEvtMsgs';
+
+/** @typedef {!Object<!string,!Array<!string>>} */
+export let AmpAnalytics3pEventMap;
```

Comment:
This was implemented this way, back when there was batching/throttling. With the wire format as it is currently, the other side (see https://github.com/google/amphtml/pull/20/files#diff-65167c4b016e67bd1ccdbbb1a3a0e99fR110) can iterate through the <transport id, events array> map, find the listener for that transport (aka creative), and then dispatch all the events for that creative in an array. In other words, the xframe implements a method which takes an array of strings.
If we change the wire format as you suggest, we have 2 choices on the 3p side:
1. Iterate through the list of {transportId: string, message: string} and for each one, look up the listener for that transportId and pass it that message. This is an API change from how it is implemented currently, but that's of course okay since we haven't launched yet.
2. Iterate through the list of {transportId: string, message: string} and rebuild {transportId: string, messages: array<string>}. Then iterate through that map and for each entry, do what we're doing now: look up the listener for that transportId and pass it the array of messages.

Either is fine with me, but you mentioned that batching/throttling is being implemented separately/externally. Do you know what that would look like? If we change the API for choice 1 that's fine, but I don't want to then change it again if we go back to something that looks similar to choice 2.

Separate from that, I completely agree that queueSize() is wrong.